src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/alerting/NewEnvelopesChecker.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.alerting;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.EnvelopeRepository;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.function.Supplier;
+
+import static java.time.DayOfWeek.SATURDAY;
+import static java.time.DayOfWeek.SUNDAY;
+import static java.time.ZoneOffset.UTC;
+import static java.util.Arrays.asList;
+
+public class NewEnvelopesChecker {
+
+    private static final Logger logger = LoggerFactory.getLogger(NewEnvelopesChecker.class);
+
+    public static final Duration TIME_WINDOW = Duration.ofHours(1);
+    public static final int START_HOUR = 10;
+    public static final int END_HOUR = 18;
+
+    private final EnvelopeRepository repo;
+    private final Supplier<LocalDateTime> timeSupplier;
+
+    public NewEnvelopesChecker(EnvelopeRepository repo, Supplier<LocalDateTime> timeSupplier) {
+        this.repo = repo;
+        this.timeSupplier = timeSupplier;
+    }
+
+    public void checkIfEnvelopesAreMissing() {
+        LocalDateTime now = timeSupplier.get();
+
+        boolean isWeekend = asList(SATURDAY, SUNDAY).contains(now.getDayOfWeek());
+        boolean isBusinessHours = START_HOUR <= now.getHour() && now.getHour() <= END_HOUR;
```

Comment:
since end hour is 18, perhaps comparison needs to be `<`?

Reply:
hmm, probably not because it passes -1 hour to repo

Reply:
18 is the last time it will run and it's checking for envelopes created after 5pm.
We get envelopes until 6pm.

Reply:
If we're making this dependent on the times the job runs, do we need `START_HOUR` and `END_HOUR`? We could just check if there are no new envelopes from the last hour.